### PR TITLE
[Layout] Use semantic tags

### PR DESCRIPTION
**Pull request by afc163**
### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

close https://github.com/ant-design/ant-design/issues/14801

I was rather worried that the tag names generated when using the <Layout> <Header> <Footer> <Content> components were nothing close to their meaning !

We have a lot of semantic tag names now available since HTML5 : `<header> <main> <article> <section> <footer>` (and so on..)

So, my proposition is to give us a way to be able to generate a truly semantic tag hierarchy using <Layout> like the proposed one : 

```html
<body>
<header>header content</header>
<main>content children</main>
<footer>page footer</footer>
</body>
```

### API Realization (Optional if not new feature)

Generate a `<header>` `<footer>`  `<main>` tag for `Header`, `Footer`, `Content`.

### What's the effect? (Optional if not new feature)

No particularly, some test snapshots may change.

### Changelog description (Optional if not new feature)

- Use semantic tags in Layout components. #14801

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

**Review by zombieJ — APPROVED** [review]

File: components/layout/layout.tsx [hunk 1]
```diff
@@ -6,13 +6,15 @@ import { ConfigConsumer, ConfigConsumerProps } from '../config-provider';
 
 export interface GeneratorProps {
   suffixCls: string;
+  tagName: 'header' | 'footer' | 'main' | 'section';
 }
 export interface BasicProps extends React.HTMLAttributes<HTMLDivElement> {
   prefixCls?: string;
   hasSider?: boolean;
+  tagName: 'header' | 'footer' | 'main' | 'section';
```
Comment (ztplz) [inline components/layout/layout.tsx]: 不optional吗

**Review by ztplz — APPROVED** [review]

File: components/layout/layout.tsx (hunk 1 above)
Comment (afc163) [reply components/layout/layout.tsx]: 内部使用，关系不大。

**Comment by adamborowski** [comment]
Hello, I think this new required parameter "tagName" is not yet documented: https://ant.design/components/layout/ Can you please update that? It would make upgrade easier.

**Comment by creeve** [comment]
Why is "tagName" required?  I don't want to go back to 50 places in my app just to change a "div" to a "section".  I don't care about which HTML tag is rendered.

**Comment by maximkott** [comment]
Why do you introduce breaking changes in a minor version change and without any real reason to do so? You do realise that anyone who is updating to the latest antd version is required to update 100 files in his project in order to add this meaningless "tagName" property?

**Comment by afc163** [comment]
@maximkott It is a bug and fixed in #15181, we will release a patch this week ASAP.

**Comment by maximkott** [comment]
Great, thanks for the info!

**Comment by afc163** [comment]
Fixed in 3.15.0

**Pull request merged**